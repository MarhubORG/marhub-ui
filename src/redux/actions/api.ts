import {
  EXPORTING_IRAP_DATA,
  EXPORT_IRAP_DATA_FAILURE,
  EXPORT_IRAP_DATA_SUCCESS,
  EXPORTING_BOTPRESS_DATA,
  EXPORT_BOTPRESS_DATA_FAILURE,
  EXPORT_BOTPRESS_DATA_SUCCESS,
  LOGOUT,
} from '../constants/actionTypes';

export interface ExportingIrapDataPayload {
  startDate: string;
  endDate: string;
  selectedTemplate: string;
  emailText: string;
  irapUuidSearchText: string;
}

export interface ExportingIrapDataAction {
  type: typeof EXPORTING_IRAP_DATA;
  payload: ExportingIrapDataPayload;
}

export function exportingIrapData(
  data: ExportingIrapDataPayload
): ExportingIrapDataAction {
  return { type: EXPORTING_IRAP_DATA, payload: data };
}

export interface ExportIrapDataSuccessAction {
  type: typeof EXPORT_IRAP_DATA_SUCCESS;
  payload: object[];
}

export function exportingIrapDataSuccess(
  data: object[]
): ExportIrapDataSuccessAction {
  return {
    type: EXPORT_IRAP_DATA_SUCCESS,
    payload: data,
  };
}

export interface ExportIrapDataFailureAction {
  type: typeof EXPORT_IRAP_DATA_FAILURE;
  payload: string;
}

export function exportingIrapDataFailure(
  message: string
): ExportIrapDataFailureAction {
  return {
    type: EXPORT_IRAP_DATA_FAILURE,
    payload: message,
  };
}

export interface ExportingBotpressDataPayload {
  startDate: string;
  endDate: string;
  selectedTemplate: string;
  emailText: string;
  irapUuidSearchText: string;
}

export interface ExportingBotpressDataAction {
  type: typeof EXPORTING_BOTPRESS_DATA;
  payload: ExportingBotpressDataPayload;
}

export function exportingBotpressData(
  data: ExportingBotpressDataPayload
): ExportingBotpressDataAction {
  return { type: EXPORTING_BOTPRESS_DATA, payload: data };
}

export interface ExportBotpressDataSuccessAction {
  type: typeof EXPORT_BOTPRESS_DATA_SUCCESS;
  payload: object[];
}

export function exportingBotpressDataSuccess(
  data: object[]
): ExportBotpressDataSuccessAction {
  return {
    type: EXPORT_BOTPRESS_DATA_SUCCESS,
    payload: data,
  };
}

export interface ExportBotpressDataFailureAction {
  type: typeof EXPORT_BOTPRESS_DATA_FAILURE;
  payload: string;
}

export function exportingBotpressDataFailure(
  message: string
): ExportBotpressDataFailureAction {
  return {
    type: EXPORT_BOTPRESS_DATA_FAILURE,
    payload: message,
  };
}

export interface LogoutAction {
  type: typeof LOGOUT;
}

export function logout(): LogoutAction {
  return { type: LOGOUT };
}

export type ApiActionTypes =
  | ExportingIrapDataAction
  | ExportIrapDataSuccessAction
  | ExportIrapDataFailureAction
  | ExportingBotpressDataAction
  | ExportBotpressDataSuccessAction
  | ExportBotpressDataFailureAction
  | LogoutAction;
