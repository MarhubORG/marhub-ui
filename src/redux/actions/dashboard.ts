import {
  FETCH_ORGANIZATIONS,
  FETCH_ORGANIZATIONS_FAILURE,
  FETCH_ORGANIZATIONS_SUCCESS,
  LOGOUT,
  UPDATE_ORGANIZATION,
  UPDATE_ORGANIZATION_FAILURE,
  UPDATE_ORGANIZATION_SUCCESS,
} from '../constants/actionTypes';

export interface FetchOrganizationsAction {
  type: typeof FETCH_ORGANIZATIONS;
}

export function fetchOrganizations(): FetchOrganizationsAction {
  return { type: FETCH_ORGANIZATIONS };
}

export interface OrganizationFields {
  name: string;
  visible_fields: string[];
}

export interface Organization {
  organisation: OrganizationFields;
  id: number;
}

export interface FetchOrganizationsSuccessAction {
  type: typeof FETCH_ORGANIZATIONS_SUCCESS;
  payload: Organization[];
}

export function fetchOrganizationsSuccess(
  organizations: Organization[]
): FetchOrganizationsSuccessAction {
  return { type: FETCH_ORGANIZATIONS_SUCCESS, payload: organizations };
}

export interface FetchOrganizationsFailureAction {
  type: typeof FETCH_ORGANIZATIONS_FAILURE;
}

export function fetchOrganizationsFailure(): FetchOrganizationsFailureAction {
  return { type: FETCH_ORGANIZATIONS_FAILURE };
}

export interface LogoutAction {
  type: typeof LOGOUT;
}

export function logout(): LogoutAction {
  return { type: LOGOUT };
}

export interface UpdateOrganizationPayload {
  id: number;
  organization: Organization;
}

export interface UpdateOrganizationAction {
  type: typeof UPDATE_ORGANIZATION;
  payload: UpdateOrganizationPayload;
}

export function updateOrganization(
  organization: UpdateOrganizationPayload
): UpdateOrganizationAction {
  return {
    type: UPDATE_ORGANIZATION,
    payload: organization,
  };
}

export interface UpdateOrganizationSuccessPayload {
  id: number;
  organization: Organization;
}

export interface UpdateOrganizationSuccessAction {
  type: typeof UPDATE_ORGANIZATION_SUCCESS;
  payload: UpdateOrganizationSuccessPayload;
}

export function updateOrganizationSuccess(
  organization: UpdateOrganizationSuccessPayload
): UpdateOrganizationSuccessAction {
  return {
    type: UPDATE_ORGANIZATION_SUCCESS,
    payload: organization,
  };
}

export interface UpdateOrganizationFailureAction {
  type: typeof UPDATE_ORGANIZATION_FAILURE;
  payload: string;
}

export function updateOrganizationFailure(
  message: string
): UpdateOrganizationFailureAction {
  return {
    type: UPDATE_ORGANIZATION_FAILURE,
    payload: message,
  };
}

export type DashboardActionTypes =
  | LogoutAction
  | FetchOrganizationsAction
  | FetchOrganizationsSuccessAction
  | FetchOrganizationsFailureAction
  | UpdateOrganizationAction
  | UpdateOrganizationSuccessAction
  | UpdateOrganizationFailureAction;
