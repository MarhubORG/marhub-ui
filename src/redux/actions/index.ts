import { SIGNUP, SIGNUP_REDIRECTING } from '../constants/actionTypes';
import { Action } from '../../types/interfaces';

export function signup(
  name: string,
  organization: string,
  email: string,
  password: string
): Action {
  return { type: SIGNUP, payload: { name, organization, email, password } };
}

export function signupRedirecting(): Action {
  return { type: SIGNUP_REDIRECTING };
}
