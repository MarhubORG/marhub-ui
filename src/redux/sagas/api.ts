import axios from 'axios';
import { put, takeLatest } from 'redux-saga/effects';
import cookie from 'react-cookies';

import { EXPORTING_IRAP_DATA } from '../constants/actionTypes';
import {
  exportingIrapDataSuccess,
  exportingIrapDataFailure,
  ExportingIrapDataAction,
} from '../actions/api';
import { formatDate } from '../../utils/excel';

function* exportIrapData(action: ExportingIrapDataAction): object {
  const formattedStartDate = formatDate(action.payload.startDate);
  const formattedEndDate = formatDate(action.payload.endDate);

  try {
    const token = cookie.load('token');
    const axiosConfig = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
      params: {
        formattedStartDate,
        formattedEndDate,
      },
    };
    const url = 'http://localhost:8080/api/v1/irap_download';

    const json = yield axios.get(url, axiosConfig);
    yield put(exportingIrapDataSuccess(json.data[0]));
  } catch (error) {
    const errorMessage = `${error.request.status} Error: ${error.response.data.error}`;
    yield put(exportingIrapDataFailure(errorMessage));
  }
}

export function* exportingIrapDataWatcher(): object {
  yield takeLatest(EXPORTING_IRAP_DATA, exportIrapData);
}
