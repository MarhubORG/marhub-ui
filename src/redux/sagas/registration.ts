import axios from 'axios';
import { put, takeLatest } from 'redux-saga/effects';
import cookie from 'react-cookies';

import { SIGNUP, LOGIN } from '../constants/actionTypes';
import {
  signupSuccess,
  signupError,
  loginError,
  loginSuccess,
  SignupAction,
  LoginAction,
} from '../actions/index';

function* signup(action: SignupAction) {
  try {
    const url = 'http://localhost:8080/api/v1/register';
    yield axios.post(url, {
      name: action.payload.name,
      organisation: action.payload.organization,
      email: action.payload.email,
      password: action.payload.password,
    });
    yield put(signupSuccess());
  } catch (error) {
    yield put(signupError(error.response.data.message));
  }
}

export function* actionWatcher() {
  yield takeLatest(SIGNUP, signup);
}

function* login(action: LoginAction) {
  try {
    const url = 'http://localhost:8080/api/v1/login';
    console.log({ url });
    const json = yield axios.post(url, {
      email: action.payload.email,
      password: action.payload.password,
    });
    console.log({ json });
    cookie.save('token', json.data.user.sessionToken, {
      path: '/',
      // domain: 'http://localhost:8080',
      // secure: true,
      // httpOnly: true,
    });
    yield put(loginSuccess());
  } catch (error) {
    yield put(loginError(error.response.data.message));
  }
}

export function* watchLogin() {
  yield takeLatest(LOGIN, login);
}
