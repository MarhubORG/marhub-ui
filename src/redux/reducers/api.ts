import { ApiState } from '../../types/interfaces';
import {
  EXPORTING_IRAP_DATA,
  EXPORT_IRAP_DATA_FAILURE,
  EXPORT_IRAP_DATA_SUCCESS,
  LOGOUT,
} from '../constants/actionTypes';
import { ApiActionTypes } from '../actions/api';

export const initialState = {
  loading: false,
  irapState: {},
  error: '',
};

export default function apiReducer(
  state = initialState,
  action: ApiActionTypes
): ApiState {
  switch (action.type) {
    case EXPORTING_IRAP_DATA:
      return { ...state, loading: true };
    case EXPORT_IRAP_DATA_SUCCESS:
      return { ...state, loading: false, irapState: action.payload };
    case EXPORT_IRAP_DATA_FAILURE:
      return { ...state, loading: false, error: action.payload };
    case LOGOUT:
      return { ...initialState };
    default:
      return state;
  }
}
