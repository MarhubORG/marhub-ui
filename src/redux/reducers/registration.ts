import { RegistrationState } from '../../types/interfaces';
import {
  SIGNUP,
  SIGNUP_SUCCESS,
  SIGNUP_ERROR,
  SIGNUP_REDIRECTING,
  LOGIN,
  LOGIN_ERROR,
  LOGIN_SUCCESS,
  LOGIN_REDIRECTING,
} from '../constants/actionTypes';
import { RegistrationActionTypes } from '../actions/index';

export function registrationReducer(
  state = initialState,
  action: RegistrationActionTypes
): RegistrationState {
  switch (action.type) {
    case SIGNUP:
      return { ...state, loading: true };
    case SIGNUP_SUCCESS:
      return { ...state, loading: false, redirect: true };
    case SIGNUP_REDIRECTING:
      return { ...state, redirect: false };
    case SIGNUP_ERROR:
      return {
        ...state,
        loading: false,
        isLoggedIn: false,
        error: action.payload.message,
      };
    case LOGIN:
      return { ...state, loading: true };
    case LOGIN_ERROR:
      return {
        ...state,
        loading: false,
        isLoggedIn: false,
        error: action.payload.message,
      };
    case LOGIN_SUCCESS:
      return {
        ...state,
        loading: false,
        loginRedirect: true,
        isLoggedIn: true,
        role: action.payload.role,
      };
    case LOGIN_REDIRECTING:
      return { ...state, loginRedirect: false };
    default:
      return state;
  }
}

export default registrationReducer;

export const initialState: RegistrationState = {
  loading: false,
  isLoggedIn: false,
  redirect: false,
  loginRedirect: false,
  error: '',
  role: '',
};
