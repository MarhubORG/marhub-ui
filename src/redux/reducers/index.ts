import { combineReducers } from 'redux';
import registration from './registration';
import apiReducer from './api';
import dashboardReducer from './dashboard';

export default combineReducers({
  registration,
  apiReducer,
  dashboardReducer,
});
