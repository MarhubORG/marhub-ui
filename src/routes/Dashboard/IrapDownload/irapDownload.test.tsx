import React from 'react';
import renderer from 'react-test-renderer';
import {
  UnconnectedIrapDownload,
  mapStateToProps,
  mapDispatchToProps,
  MapDispatchToProps,
} from './irapDownload';
import { exportingIrapData } from '../../../redux/actions/api';
import { initialState } from '../../../redux/reducers/registration';

it('renders correctly when logged in', () => {
  const tree = renderer
    .create(
      <UnconnectedIrapDownload
        isLoggedIn
        exportingIrapData={exportingIrapData}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when not logged in', () => {
  const tree = renderer
    .create(
      <UnconnectedIrapDownload
        isLoggedIn
        exportingIrapData={exportingIrapData}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('mapStateToProps', () => {
  it('contains registration state', () => {
    expect(mapStateToProps({ registration: initialState })).toBeDefined();
    expect(mapStateToProps({ registration: initialState }).isLoggedIn).toEqual(
      false
    );
  });
});

describe('mapDispatchToProps', () => {
  it('contains a function called exportingIrapData', () => {
    const dispatch = jest.fn();
    const dispatches: MapDispatchToProps = mapDispatchToProps(dispatch);
    expect(Object.keys(dispatches).includes('exportingIrapData')).toBeTruthy();
    expect(typeof dispatches.exportingIrapData).toEqual('function');
    expect(dispatches.exportingIrapData.name).toEqual('exportingIrapData');
    expect(dispatches.exportingIrapData()).toBeUndefined();
  });
});
