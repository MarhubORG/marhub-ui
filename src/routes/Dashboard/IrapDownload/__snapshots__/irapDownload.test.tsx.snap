// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when logged in 1`] = `
.c2 {
  margin: 1rem;
  border-radius: 2rem;
  border: 0px red;
  padding: 0.5rem 1rem;
}

.c2 .react-date-picker__wrapper {
  border: 0px;
}

.c2 .react-date-picker__wrapper {
  border: 0px;
}

.c2 .react-date-picker__wrapper {
  border: 0px;
}

.c2 .react-date-picker__wrapper {
  border: 0px;
}

.c0 {
  margin-top: 2rem;
}

.c3 {
  margin: 1rem;
  width: 12rem;
  height: 3rem;
  font-family: Open Sans,sans-serif;
  font-weight: 700;
  border-radius: 0.2rem;
  font-size: 0.8rem;
}

.c1 {
  display: block;
  margin-left: 4.7rem;
}

<div
  className="c0"
>
  <div>
    <label
      className="c1"
    >
      Start date:
    </label>
    <div
      className="react-date-picker react-date-picker--closed react-date-picker--enabled c2"
      onFocus={[Function]}
    >
      <div
        className="react-date-picker__wrapper"
      >
        <div
          className="react-date-picker__inputGroup"
          onClick={[Function]}
        >
          <input
            max="5760-09-12"
            min="1821-04-19"
            name="date"
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "left": "-9999px",
                "position": "absolute",
                "top": "-9999px",
                "visibility": "hidden",
              }
            }
            type="date"
            value="2020-04-01"
          />
          
          <input
            autoComplete="off"
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
            max={12}
            min={1}
            name="month"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="--"
            required={null}
            type="number"
            value={4}
          />
          <span
            className="react-date-picker__inputGroup__divider"
          >
            /
          </span>
          <input
            autoComplete="off"
            autoFocus={false}
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
            max={30}
            min={1}
            name="day"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="--"
            required={null}
            type="number"
            value={1}
          />
          <span
            className="react-date-picker__inputGroup__divider"
          >
            /
          </span>
          <input
            autoComplete="off"
            autoFocus={false}
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
            max={275760}
            min={1000}
            name="year"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="----"
            required={null}
            step={1}
            type="number"
            value={2020}
          />
          
        </div>
        <button
          className="react-date-picker__clear-button react-date-picker__button"
          onClick={[Function]}
          onFocus={[Function]}
          type="button"
        >
          <svg
            className="react-date-picker__clear-button__icon react-date-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="4"
              x2="15"
              y1="4"
              y2="15"
            />
            <line
              x1="15"
              x2="4"
              y1="4"
              y2="15"
            />
          </svg>
        </button>
        <button
          className="react-date-picker__calendar-button react-date-picker__button"
          onClick={[Function]}
          onFocus={[Function]}
          type="button"
        >
          <svg
            className="react-date-picker__calendar-button__icon react-date-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="15"
              width="15"
              x="2"
              y="2"
            />
            <line
              x1="6"
              x2="6"
              y1="0"
              y2="4"
            />
            <line
              x1="13"
              x2="13"
              y1="0"
              y2="4"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div>
    <label
      className="c1"
    >
      End date:
    </label>
    <div
      className="react-date-picker react-date-picker--closed react-date-picker--enabled c2"
      onFocus={[Function]}
    >
      <div
        className="react-date-picker__wrapper"
      >
        <div
          className="react-date-picker__inputGroup"
          onClick={[Function]}
        >
          <input
            max="5760-09-12"
            min="1821-04-19"
            name="date"
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "left": "-9999px",
                "position": "absolute",
                "top": "-9999px",
                "visibility": "hidden",
              }
            }
            type="date"
            value="2020-04-08"
          />
          
          <input
            autoComplete="off"
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
            max={12}
            min={1}
            name="month"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="--"
            required={null}
            type="number"
            value={4}
          />
          <span
            className="react-date-picker__inputGroup__divider"
          >
            /
          </span>
          <input
            autoComplete="off"
            autoFocus={false}
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
            max={30}
            min={1}
            name="day"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="--"
            required={null}
            type="number"
            value={8}
          />
          <span
            className="react-date-picker__inputGroup__divider"
          >
            /
          </span>
          <input
            autoComplete="off"
            autoFocus={false}
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
            max={275760}
            min={1000}
            name="year"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="----"
            required={null}
            step={1}
            type="number"
            value={2020}
          />
          
        </div>
        <button
          className="react-date-picker__clear-button react-date-picker__button"
          onClick={[Function]}
          onFocus={[Function]}
          type="button"
        >
          <svg
            className="react-date-picker__clear-button__icon react-date-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="4"
              x2="15"
              y1="4"
              y2="15"
            />
            <line
              x1="15"
              x2="4"
              y1="4"
              y2="15"
            />
          </svg>
        </button>
        <button
          className="react-date-picker__calendar-button react-date-picker__button"
          onClick={[Function]}
          onFocus={[Function]}
          type="button"
        >
          <svg
            className="react-date-picker__calendar-button__icon react-date-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="15"
              width="15"
              x="2"
              y="2"
            />
            <line
              x1="6"
              x2="6"
              y1="0"
              y2="4"
            />
            <line
              x1="13"
              x2="13"
              y1="0"
              y2="4"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <button
    className="c3"
    onClick={[Function]}
  >
    Download IRAP Data
  </button>
</div>
`;
